### PR TITLE
Create Train Description Program and Passenger Struct

### PR DESCRIPTION
**Pull request by robskrob**

### CODE REVIEW

File: train.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+destination = {}
+destination[:current_city] = 'New York'
+
+
+Passenger = Struct.new(:name, :destination, :description)
+bill = Passenger.new({name: 'Bill Smith'}, {destination: 'New York'}, {number_of_engines: 4, caboose: true, number_of_cars: 5})
+
+#Train = Struct.new(:destination, :description)
+#train = Train.new( {destination: 'New York'}, {number_of_engines: 4, caboose: true, number_of_cars: 5})
```
Comment (jwo) [inline train.rb]: What I was going for was 

``` ruby
passengers = [Passenger.new(name: 'Bill Smith'), Passenger.new(name: 'Sarah Robot')]
Train.new( passengers, {destination: 'New York', number_of_engines: 4, caboose: true, number_of_cars: 5})
```

What would have to change to be able to use this interface?

**Comment by jwo** [comment]
I don't think the following is the best API for usage

```
destination.each do |key, value|
  puts "#{key}: #{value}"
end
```

Instead, I was looking for:

```
puts "Destination: #{train.destination}"
puts "Cars: #{train.number_of_cars}"
```

**Comment by robskrob** [comment]
Thank you for your feed back. I will look into this within the next 12
hours and get back to you with my questions/corrections.

On Mon, Jan 13, 2014 at 10:24 AM, Jesse Wolgamott
notifications@github.comwrote:

> I don't think the following is the best API for usage
> 
> destination.each do |key, value|
>   puts "#{key}: #{value}"
> end
> 
> Instead, I was looking for:
> 
> puts "Destination: #{train.destination}"
> puts "Cars: #{train.number_of_cars}"
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/RubyoffRails/recipe-hash/pull/13#issuecomment-32178536
> .

## 

Robert Jewell
1.914.400.5219

File: train.rb (hunk 1 above)
Comment (robskrob) [inline train.rb]: I'm sorry, but I am a little lost here. I also do not even know if I am participating in this thread correctly. How do I insert syntax colors like you did above?

In order to use what you were going for, I think you must have two struct variables 

Passenger = Struct.new(:name)
Train = Struct.new(:destination, :number_of_cars, :caboose, :number_of_engine)

And then with those struct variables, you create struct objects:
passengers = [Passenger.new(name: 'Bill Smith'), Passenger.new(name: 'Sarah Robot')]
train = Train.new( passengers, {destination: 'New York', number_of_engines: 4, caboose: true, number_of_cars: 5})

Then with the struct objects you print the information:

puts "Destination: #{train.destination}"
puts "Cars: #{train.number_of_cars}"
puts "Caboose: #{train.caboose}"
puts "Passengers: #{train.passengers}"
puts "Engines: #{train.number_of_engines}"

**Comment by robskrob** [comment]
Hey Jesse, I responded to you in Git, but I do not think I followed
standard protocol.

On Mon, Jan 13, 2014 at 7:21 PM, Robert Jewell jewell.robertp@gmail.comwrote:

> Thank you for your feed back. I will look into this within the next 12
> hours and get back to you with my questions/corrections.
> 
> On Mon, Jan 13, 2014 at 10:24 AM, Jesse Wolgamott <
> notifications@github.com> wrote:
> 
> > I don't think the following is the best API for usage
> > 
> > destination.each do |key, value|
> >   puts "#{key}: #{value}"
> > end
> > 
> > Instead, I was looking for:
> > 
> > puts "Destination: #{train.destination}"
> > puts "Cars: #{train.number_of_cars}"
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/RubyoffRails/recipe-hash/pull/13#issuecomment-32178536
> > .
> 
> ## 
> 
> Robert Jewell
> 1.914.400.5219

## 

Robert Jewell
1.914.400.5219

**Comment by jwo** [comment]
@rjewell2200 Did you push your changes up to github? That's the key for me to see it here.

**Comment by robskrob** [comment]
Ok I will now

On Tue, Jan 14, 2014 at 11:51 AM, Jesse Wolgamott
notifications@github.comwrote:

> @rjewell2200 https://github.com/rjewell2200 Did you push your changes
> up to github? That's the key for me to see it here.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/RubyoffRails/recipe-hash/pull/13#issuecomment-32282768
> .

## 

Robert Jewell
1.914.400.5219

**Comment by robskrob** [comment]
I created two new structs

**Comment by jwo** [comment]
I don't think you meant to close the PR thought, right?

**Comment by robskrob** [comment]
I'm sorry, I did not mean that. I feel like i'm fumbling about git

On Tue, Jan 14, 2014 at 12:25 PM, Jesse Wolgamott
notifications@github.comwrote:

> I don't think you meant to close the PR thought, right?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/RubyoffRails/recipe-hash/pull/13#issuecomment-32286389
> .

## 

Robert Jewell
1.914.400.5219

**Pull request closed without merging**